sound/soc/sof/intel/hda-codec.c

```diff
@@ -200,7 +203,8 @@ int hda_codec_i915_exit(struct snd_sof_dev *sdev)
 	struct hdac_bus *bus = sof_to_bus(sdev);
 	int ret;
 
-	hda_codec_i915_display_power(sdev, false);
+	/* power down unconditionally */
+	snd_hdac_display_power(bus, HDA_CODEC_IDX_CONTROLLER, false);
```

Comment:
sorry, I am confused. In parts of this patch you replace hda_codec_i915_display_power() by snd_hdac_display_power(), but in other parts you keep the hda_codec_i915_display_power() ?

Reply:
@plbossart I now moved the checks for codec_mask from callers (which are many now) to  hda_codec_i915_display_power() directly. There are only two calls at init where we want to unconditionally change power (above code is one). And in these places I call snd_hdac_display_power() directly. E.g. at probe, codec_mask is not yet set, so we have to power on for the probe.